sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -296,6 +296,24 @@ class Column(protected[sql] val expr: Expression) extends Logging {
   def eqNullSafe(other: Any): Column = this <=> other
 
   /**
+   * True if the current column is between the lower bound and upper bound, inclusive.
+   *
+   * @group java_expr_ops
+   */
+  def between(lowerBound: String, upperBound: String): Column = {
```

Comment:
i think you can remove this function, and change the signature of the one below to take "Any".